src/electron/flux/action-creator/scan-action-creator.ts

```diff
@@ -1,11 +1,16 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { DeviceActions } from 'electron/flux/action/device-actions';
 import { ScanActions } from 'electron/flux/action/scan-actions';
 
 export class ScanActionCreator {
-    constructor(private readonly scanActions: ScanActions) {}
+    constructor(
+        private readonly scanActions: ScanActions,
+        private readonly deviceActions: DeviceActions,
+    ) {}
 
     public scan(port: number): void {
+        this.deviceActions.resetConnection.invoke(null);
```

Comment:
Do we need the `null` parameter here? I see it's also passed by `DeviceConnectActionCreator` but it doesn't seem to be used?

Reply:
make sense